### PR TITLE
Site to Site VPN logging

### LINKED ISSUE #26389: Site-to-Site VPN Tunnel logs
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS recently enabled sending the VPN Tunnel logs to Cloudwatch. This request is to include this feature in the AWS Terraform provider.

### New or Affected Resource(s)

* `aws_vpn_connection`

[Announcement](https://aws.amazon.com/about-aws/whats-new/2022/08/aws-site-vpn-connection-logs-amazon-cloudwatch/).

### PR DESCRIPTION
**Pull request by bschaatsbergen**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #26389

Output from acceptance testing:

```
$ make testacc TESTS=TestAccXXX PKG=ec2

...
```

### CODE REVIEW

File: internal/service/ec2/vpnsite_connection.go [hunk 1]
```diff
@@ -189,6 +189,38 @@ func ResourceVPNConnection() *schema.Resource {
...
+									},
+									"log_group_arn": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"log_output_format": {
+										Type:     schema.TypeString,
+										Optional: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											"json", "text",
```
Comment (ewbankkit) [inline internal/service/ec2/vpnsite_connection.go:215]: Please add these (and a corresponding `_Values()` function) to `internal/service/ec2/consts.go`.

File: internal/service/ec2/vpnsite_connection.go [hunk 2]
```diff
@@ -189,6 +189,38 @@ func ResourceVPNConnection() *schema.Resource {
 					ValidateFunc: validation.StringInSlice(vpnTunnelOptionsIKEVersion_Values(), false),
 				},
 			},
+			"tunnel1_log_options": {
+				Type:     schema.TypeSet,
```
Comment (ewbankkit) [inline internal/service/ec2/vpnsite_connection.go:193]:
```suggestion
				Type:     schema.TypeList,
				MaxItems:     1,
```

as there is at most a single `VpnTunnelLogOptionsSpecification`.

File: internal/service/ec2/vpnsite_connection.go [hunk 3]
```diff
@@ -960,6 +1024,10 @@ func expandVPNTunnelOptionsSpecification(d *schema.ResourceData, prefix string)
 		}
 	}
 
+	if v, ok := d.GetOk(prefix + "log_options"); ok && v.(*schema.Set).Len() > 0 {
+		apiObject.LogOptions = expandVPNTunnelLogOptionsSpecification(d)
+	}
```
Comment (ewbankkit) [inline internal/service/ec2/vpnsite_connection.go:1029]:
```suggestion
	if v, ok := d.GetOk(prefix + "log_options"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
		apiObject.LogOptions = expandVPNTunnelLogOptionsSpecification(v.([]interface{})[0].(map[string]interface{}))
	}
```

and change parameter of `expandVPNTunnelLogOptionsSpecification` to `tfMap map[string]interface{}`.

Similar on for _flatten_.

File: internal/service/ec2/vpnsite_connection.go (hunk 1 above)
Comment (bschaatsbergen) [reply internal/service/ec2/vpnsite_connection.go:215]: Fixed :+1:

File: internal/service/ec2/vpnsite_connection.go (hunk 2 above)
Comment (bschaatsbergen) [reply internal/service/ec2/vpnsite_connection.go:193]: Fixed :+1:

File: website/docs/r/vpn_connection.html.markdown [hunk 4]
```diff
@@ -181,6 +183,20 @@ Other arguments:
 * `tunnel1_startup_action` - (Optional, Default `add`) The action to take when the establishing the tunnel for the first VPN connection. By default, your customer gateway device must initiate the IKE negotiation and bring up the tunnel. Specify start for AWS to initiate the IKE negotiation. Valid values are `add | start`.
 * `tunnel2_startup_action` - (Optional, Default `add`) The action to take when the establishing the tunnel for the second VPN connection. By default, your customer gateway device must initiate the IKE negotiation and bring up the tunnel. Specify start for AWS to initiate the IKE negotiation. Valid values are `add | start`.
 
+### Log Options
+
+The `tunnel1_log_options` and `tunnel2_log_options` block supports the following arguments:
+
+* `cloudwatch_log_options` - (Required) Options for sending VPN tunnel logs to CloudWatch. See [CloudWatch Log Options](#cloudwatch-log-options) below for more details.
```
Comment (tmatilai) [inline website/docs/r/vpn_connection.html.markdown:190]: This should be documented as optional, right?

Comment (ewbankkit) [reply website/docs/r/vpn_connection.html.markdown]: 👍

**Comment by ewbankkit** [comment]
#### TODO

Test log options update in `TestAccSiteVPNConnection_cloudWatchLogOptions`.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% ACCTEST_TIMEOUT=360m make testacc TESTARGS='-run=TestAccSiteVPNConnection_' PKG=ec2 ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 2  -run=TestAccSiteVPNConnection_ -timeout 360m
=== RUN   TestAccSiteVPNConnection_basic
=== PAUSE TestAccSiteVPNConnection_basic
=== RUN   TestAccSiteVPNConnection_cloudWatchLogOptions
=== PAUSE TestAccSiteVPNConnection_cloudWatchLogOptions
=== RUN   TestAccSiteVPNConnection_transitGatewayID
=== PAUSE TestAccSiteVPNConnection_transitGatewayID
=== RUN   TestAccSiteVPNConnection_tunnel1InsideCIDR
=== PAUSE TestAccSiteVPNConnection_tunnel1InsideCIDR
=== RUN   TestAccSiteVPNConnection_tunnel1InsideIPv6CIDR
=== PAUSE TestAccSiteVPNConnection_tunnel1InsideIPv6CIDR
=== RUN   TestAccSiteVPNConnection_tunnel1PreSharedKey
=== PAUSE TestAccSiteVPNConnection_tunnel1PreSharedKey
=== RUN   TestAccSiteVPNConnection_tunnelOptions
=== PAUSE TestAccSiteVPNConnection_tunnelOptions
=== RUN   TestAccSiteVPNConnection_tunnelOptionsLesser
=== PAUSE TestAccSiteVPNConnection_tunnelOptionsLesser
=== RUN   TestAccSiteVPNConnection_staticRoutes
=== PAUSE TestAccSiteVPNConnection_staticRoutes
=== RUN   TestAccSiteVPNConnection_outsideAddressTypePrivate
=== PAUSE TestAccSiteVPNConnection_outsideAddressTypePrivate
=== RUN   TestAccSiteVPNConnection_outsideAddressTypePublic
=== PAUSE TestAccSiteVPNConnection_outsideAddressTypePublic
=== RUN   TestAccSiteVPNConnection_enableAcceleration
=== PAUSE TestAccSiteVPNConnection_enableAcceleration
=== RUN   TestAccSiteVPNConnection_ipv6
=== PAUSE TestAccSiteVPNConnection_ipv6
=== RUN   TestAccSiteVPNConnection_tags
=== PAUSE TestAccSiteVPNConnection_tags
=== RUN   TestAccSiteVPNConnection_specifyIPv4
=== PAUSE TestAccSiteVPNConnection_specifyIPv4
=== RUN   TestAccSiteVPNConnection_specifyIPv6
=== PAUSE TestAccSiteVPNConnection_specifyIPv6
=== RUN   TestAccSiteVPNConnection_disappears
=== PAUSE TestAccSiteVPNConnection_disappears
=== RUN   TestAccSiteVPNConnection_updateCustomerGatewayID
=== PAUSE TestAccSiteVPNConnection_updateCustomerGatewayID
=== RUN   TestAccSiteVPNConnection_updateVPNGatewayID
=== PAUSE TestAccSiteVPNConnection_updateVPNGatewayID
=== RUN   TestAccSiteVPNConnection_updateTransitGatewayID
=== PAUSE TestAccSiteVPNConnection_updateTransitGatewayID
=== RUN   TestAccSiteVPNConnection_vpnGatewayIDToTransitGatewayID
=== PAUSE TestAccSiteVPNConnection_vpnGatewayIDToTransitGatewayID
=== RUN   TestAccSiteVPNConnection_transitGatewayIDToVPNGatewayID
=== PAUSE TestAccSiteVPNConnection_transitGatewayIDToVPNGatewayID
=== CONT  TestAccSiteVPNConnection_basic
=== CONT  TestAccSiteVPNConnection_enableAcceleration
--- PASS: TestAccSiteVPNConnection_basic (198.23s)
=== CONT  TestAccSiteVPNConnection_tunnelOptions
--- PASS: TestAccSiteVPNConnection_tunnelOptions (200.80s)
=== CONT  TestAccSiteVPNConnection_outsideAddressTypePublic
--- PASS: TestAccSiteVPNConnection_enableAcceleration (409.53s)
=== CONT  TestAccSiteVPNConnection_outsideAddressTypePrivate
--- PASS: TestAccSiteVPNConnection_outsideAddressTypePublic (195.64s)
=== CONT  TestAccSiteVPNConnection_staticRoutes
--- PASS: TestAccSiteVPNConnection_staticRoutes (199.24s)
=== CONT  TestAccSiteVPNConnection_tunnelOptionsLesser
--- PASS: TestAccSiteVPNConnection_outsideAddressTypePrivate (1319.21s)
=== CONT  TestAccSiteVPNConnection_tunnel1InsideCIDR
--- PASS: TestAccSiteVPNConnection_tunnel1InsideCIDR (193.07s)
=== CONT  TestAccSiteVPNConnection_tunnel1PreSharedKey
--- PASS: TestAccSiteVPNConnection_tunnelOptionsLesser (1212.31s)
=== CONT  TestAccSiteVPNConnection_tunnel1InsideIPv6CIDR
--- PASS: TestAccSiteVPNConnection_tunnel1PreSharedKey (194.99s)
=== CONT  TestAccSiteVPNConnection_transitGatewayID
--- PASS: TestAccSiteVPNConnection_tunnel1InsideIPv6CIDR (402.33s)
=== CONT  TestAccSiteVPNConnection_updateCustomerGatewayID
--- PASS: TestAccSiteVPNConnection_transitGatewayID (416.89s)
=== CONT  TestAccSiteVPNConnection_transitGatewayIDToVPNGatewayID
--- PASS: TestAccSiteVPNConnection_updateCustomerGatewayID (547.68s)
=== CONT  TestAccSiteVPNConnection_vpnGatewayIDToTransitGatewayID
--- PASS: TestAccSiteVPNConnection_transitGatewayIDToVPNGatewayID (630.40s)
=== CONT  TestAccSiteVPNConnection_updateTransitGatewayID
--- PASS: TestAccSiteVPNConnection_vpnGatewayIDToTransitGatewayID (747.54s)
=== CONT  TestAccSiteVPNConnection_updateVPNGatewayID
--- PASS: TestAccSiteVPNConnection_updateTransitGatewayID (776.53s)
=== CONT  TestAccSiteVPNConnection_tags
--- PASS: TestAccSiteVPNConnection_updateVPNGatewayID (429.76s)
=== CONT  TestAccSiteVPNConnection_disappears
--- PASS: TestAccSiteVPNConnection_tags (222.85s)
=== CONT  TestAccSiteVPNConnection_cloudWatchLogOptions
--- PASS: TestAccSiteVPNConnection_disappears (194.64s)
=== CONT  TestAccSiteVPNConnection_specifyIPv6
--- PASS: TestAccSiteVPNConnection_specifyIPv6 (445.21s)
=== CONT  TestAccSiteVPNConnection_ipv6
--- PASS: TestAccSiteVPNConnection_cloudWatchLogOptions (622.32s)
=== CONT  TestAccSiteVPNConnection_specifyIPv4
--- PASS: TestAccSiteVPNConnection_ipv6 (398.31s)
--- PASS: TestAccSiteVPNConnection_specifyIPv4 (509.71s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	5299.536s
```

**Comment by ewbankkit** [comment]
@bschaatsbergen Thanks for the contribution :tada: :clap:.

**Pull request merged**